File: client/src/features/tenants/dto/response/tenant.response.ts
import type { UserBriefResponse } from "../../../../shared/dto/response";

export type TenantStatus = "ACTIVE" | "INACTIVE";

export interface TenantResponse {
  id: string;
  name: string;
  slug: string;
  status: TenantStatus;
  adminEmails: string[];
  createdBy: UserBriefResponse;
  updatedBy: UserBriefResponse;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTenantResponse extends TenantResponse {
  admin: {
    id: string;
    email: string;
    displayName: string;
    tenantAdmin: boolean;
  };
  temporaryPassword: string;
}

File: backend/sentinel-api/src/main/java/com/sentinel/api/tenant/dto/response/TenantResponse.java
package com.sentinel.api.tenant.dto.response;

import com.sentinel.api.common.dto.response.UserBriefResponse;
import com.sentinel.common.postgresql.tenant.TenantStatus;

import java.time.Instant;
import java.util.List;

public record TenantResponse(
    String id,
    String name,
    String slug,
    TenantStatus status,
    List<String> adminEmails,
    UserBriefResponse createdBy,
    UserBriefResponse updatedBy,
    Instant createdAt,
    Instant updatedAt) {
}
